### PR TITLE
Add V2 services/connectors support with V1 backward compatibility

### PR DESCRIPTION
**Pull request by bashinate-trustgrid-1982**
Resolves #247.

Adds full V2 services/connectors support (cluster and node) while keeping V1 customers working through the migration window.

## Scope

**New resources** (V2):
- `tg_cluster_service` / `tg_node_service`
- `tg_cluster_connector` / `tg_node_connector`
- `tg_cluster_services_v2_upgrade` / `tg_cluster_connectors_v2_upgrade` / `tg_node_services_v2_upgrade` / `tg_node_connectors_v2_upgrade` (one-shot upgrade triggers; destroy is a no-op)

**New data sources** (for ID discovery during migration):
- `tg_cluster_services` / `tg_node_services`
- `tg_cluster_connectors` / `tg_node_connectors`

**V2-only fields** on the new service resources:
- `source_interface`
- `source_from_cluster_ip` (cluster only; cross-field validator rejects `source_from_cluster_ip = true` without `source_interface` at plan time)

**Dual-shape decoder** on `tg.ServicesConfig` and `tg.ConnectorsConfig` — accepts both V1 array (`{"services":[...]}`) and V2 items map (`{"items":{id:{...}}}`) shapes returned by `/cluster/{fqdn}` and `/node/{id}`. Single provider build supports V1-only, V2-only, and mixed estates simultaneously. Marshal stays V1 so legacy `tg_service` / `tg_connector` continue working against V1 clusters.

**Importers** on every new V2 resource (format: `{target_id_or_fqdn}:{resource_id}`).

**Customer-facing migration guide** at `templates/guides/v1-to-v2-migration.md` covering the two-apply workflow, Terraform 1.7+ vs 1.5–1.6 paths, and the UUID rekey gotcha.

### Unit tests
`tg/node_test.go` (new) — 15 tests covering:
- V1 array shape decode → slice
- V2 items map decode → slice (sorted by ID, map-key fallback for missing inner ID)
- Empty/null edge cases for both shapes
- V1+V2 shapes present simultaneously (V2 preferred)
- Marshal still emits V1 array
- V2 fields omitempty on V1 payloads
- **PR #242 regression**: V2-shape `Cluster` body must NOT yield empty services. Pins the destructive scenario the closed PR would have caused.

### Acceptance tests
Six new files in `acctests/` covering the new resources and data sources, following the existing fixture-based convention (`testClusterFQDN`, `testNodeID`):
- `cluster_service_v2_test.go` — CRUD across 3 steps, validator-error case, API-side check
- `node_service_v2_test.go` — CRUD with optional `source_interface`
- `cluster_connector_v2_test.go` — CRUD with `rate_limit` update
- `node_connector_v2_test.go` — same for node
- `v2_upgrade_test.go` — idempotent re-apply, no-op destroy
- `data_sources_v2_test.go` — each of the four new data sources

Sweepers included for every new V2 resource. CI's `make sweep testacc` picks them up automatically.

⚠️ **Important for first CI run**: the V2 acctests assume `testClusterFQDN` and `testNodeID` are already V2-upgraded in the dev tenant. If they're still V1, the V2 CRUD tests will 4xx on POST. A one-time portal upgrade on the dev test cluster + node is required before this PR's CI passes.

### Live end-to-end test
Run against fresh `l4-connector` labinator env on `nate.test.trustgrid.io`. Validated the full customer journey:

| Phase | Action | Result |
|---|---|---|
| V1 setup | New provider creates V1 `tg_service` + `tg_connector` on V1 cluster | ✅ Backward compat |
| V2 CRUD (from scratch) | Create + Update + Delete on `tg_cluster_service` (with V2-only fields) and `tg_cluster_connector` (with `rate_limit`) | ✅ |
| Cross-field validator | `source_from_cluster_ip=true` without `source_interface` | ✅ Rejected at plan time |
| Migration apply 1 | `removed{}` blocks for V1 resources + `tg_cluster_{services,connectors}_v2_upgrade` | ✅ V1 forgotten, cluster upgraded |
| Migration apply 2 | `data` source + `import{}` with `for_each` (TF 1.7+) + new V2 resource blocks | ✅ 2 imported, 0 added/changed |
| Post-migration plan | `terraform plan` | ✅ "No changes" |
| Idempotent reads | Mixed estate (one V1 cluster, one V2 cluster) | ✅ Both decoded cleanly via same provider build |

### Key findings

1. **V2 upgrade endpoint rekeys all service/connector UUIDs.** Verified — service ID `4dd993c1-…` became `e805876a-…` immediately after `POST /v2/.../config/services/upgrade`. Customer migration is therefore two applies: forget V1 + upgrade in apply 1; discover new IDs + import in apply 2. Documented in the guide.
2. **Single-apply migration is structurally impossible**, even on latest Terraform (1.15.2). `import {}` block IDs / `for_each` must be known at plan time, but new V2 IDs only exist after the upgrade runs during apply. Not a version limitation.
3. **Two-apply on Terraform 1.7+ requires zero manual UUID copy-paste.** `import {}` with `for_each` referencing the data source auto-discovers rekeyed IDs by name/port. On 1.5–1.6, customer captures `terraform output` between applies and pastes literal IDs.

## Test plan
- [x] Unit tests pass locally (`go test ./...`)
- [x] Build clean (`go build ./...`)
- [x] Live V2 CRUD validated against `v2tf-…nate.test.trustgrid.io` lab env
- [x] Live V1→V2 migration validated end-to-end with `terraform plan` clean after
- [x] `make docs` produces a clean diff
- [ ] CI `make sweep testacc` passes — requires dev tenant test cluster + node to be V2-upgraded (one-time portal action)

## Out of scope (separate tickets recommended)
- Legacy `tg_connector` 422-on-destroy surfaced during testing (sibling connector with empty `nic` blocks the V1 PUT-back). Pre-existing; not introduced by this branch.
- Removing V1 cluster/node code paths — follow-up major-version bump after customer migration completes.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: resource/cluster_service.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+// clusterServiceValidate rejects source_from_cluster_ip=true unless
+// source_interface is also set. Enforced at plan time so the user sees the
+// error before any API write.
+func clusterServiceValidate(_ context.Context, d *schema.ResourceDiff, _ any) error {
+	source, _ := d.GetOk("source_interface")
+	useClusterIP, _ := d.GetOk("source_from_cluster_ip")
+	if useClusterIP == true && source == "" {
+		return fmt.Errorf("source_from_cluster_ip = true requires source_interface to be set")
+	}
```
[the remark this answers is not in the document]
Comment (bashinate-trustgrid-1982) [reply resource/cluster_service.go:27]: Good catch on the brittleness. Quick technical note: `d.GetOk` on a `TypeString` returns `("", false)` when unset (zero value of the type, not `nil`), so the original `source == ""` comparison did actually fire correctly — verified live via tcpdump where 'source_from_cluster_ip=true without source_interface' was rejected at plan time. But the typed-assertion form is unambiguously clearer and doesn't rely on interface-equality semantics. Fixed in 0798e0f.

**Review by terrbear — APPROVED** [review]
